functions/linear_section.h

```diff
@@ -0,0 +1,36 @@
+#ifndef FUNCTION_LINEAR_SECTION_H
+#define FUNCTION_LINEAR_SECTION_H
+
+#include <algorithm>
+
+// Usage: LinearSectionF<POSITION, FRACTION>
+// POSITION: FUNCTION position on the blade, 0-32768
+// FRACTION: FUNCTION how much of the blade to light up, 0 = none
+// return value: FUNCTION
+// creates a "block" of pixels at POSITION taking up FRACTION of blade
+
+class BladeBase;
+template<class POSITION, class FRACTION>
+class LinearSectionF {
+public:
+  FunctionRunResult run(BladeBase* base) {
+    pos_.run(base);
+    FunctionRunResult ret = RunFunction(&fraction_, base);
+    uint32_t num_leds_;
```

Comment:
since this is now a local variable, it should not have any _ at the end.
Also, you can merge this with the next line into:
int num_leds = base->num_leds();

Also, I don't think this compiles right now because some of the stuff below uses "num_leds" which doesn't exist right now.

Reply:
local had num_leds_, fixed on git and combined